cfripper/rules/wildcard_principals.py

```diff
@@ -104,15 +102,16 @@ def check_for_wildcards(
         extras: Optional[Dict] = None,
     ):
         for statement in resource.statement_as_list():
-            if statement.Effect == "Allow" and statement.principals_with(self.FULL_REGEX):
-                for principal in statement.get_principal_list():
-                    account_id_match = self.IAM_PATTERN.match(principal) or self.AWS_ACCOUNT_ID_PATTERN.match(principal)
-                    account_id = account_id_match.group(1) if account_id_match else None
+            filtered_principals = statement.principals_with(self.FULL_REGEX)
+            if statement.Effect == "Allow" and filtered_principals:
+                for principal in filtered_principals:
```

Comment:
Previously we iterated over all principals, but we should iterate just over the ones that match the principal